src/loaders/agent-base.js

```diff
@@ -3,9 +3,13 @@
 import { warn } from '../common/util/console'
 
 export class AgentBase {
-  /** Generates a generic warning message with the api name injected */
-  #warnMessage (api) {
-    return `Call to agent api ${api} failed. The agent is not currently initialized.`
+  /** Tries to execute the api and enerates a generic warning message with the api name injected if unsuccessful */
+  #try (method, ...args) {
+    try {
+      this.api[method](...args)
+    } catch (err) {
+      warn(`Call to agent api ${method} failed. The agent is not currently initialized.`)
```

Comment:
The try catch is assuming any exception is indicative of the agent not being fully initialized. What happens when one of the API methods throws an exception for a different reason? It may be better to change this to specifically test for the existence of the method?

Reply:
good call, will change